source/extensions/network/dns_resolver/cares/dns_impl.cc

```diff
@@ -208,6 +215,11 @@ void DnsResolverImpl::AddrInfoPendingResolution::onAresGetAddrInfoCallback(
 
     ASSERT(addrinfo != nullptr);
     ares_freeaddrinfo(addrinfo);
+  } else if (accept_nodata_ && isResponseWithNoRecords(status)) {
```

Comment:
Can you add a test please, for dual resolutions that if one resolution returns an error and the second is no data, the overall result is no data?

Reply:
sure, added some tests for this.